crates/wasmtime/src/trap.rs

```diff
@@ -98,7 +98,9 @@ pub(crate) fn from_runtime_box(
             error,
             needs_backtrace,
         } => {
-            debug_assert!(needs_backtrace == backtrace.is_some());
+            debug_assert!(
+                needs_backtrace == backtrace.is_some() || !store.engine().config().wasm_backtrace
+            );
```

Comment:
Shouldn't we just set `needs_backtrace = false` when the store has backtraces disabled, rather than having two places we need to check for consistency here?

Reply:
I agree yeah it would be good but it's not trivial at least for now. Some refactoring could probably fix this but it doesn't seem too worth it at this time so I'm going to go ahead and merge and leave this for a future rainy day